SEP-0011.md

```diff
@@ -0,0 +1,154 @@
+There are other more subtle differences and SunPy contributors find this confusing.
+This has come up in the past.
+
+While there are multiple code formatters for Python, the *Black* Python code formatter produces good enough results that it is getting significant traction in many open source projects, including: scikit-learn, pytest, tox, Pyramid, Django, Hypothesis, SQLAlchemy, virtualenv, pandas, Pillow, and Twisted.
+While adoption by this list of high-profile community-based open source projects does not automatically imply that it is a good fit for SunPy, it demonstrates unequivocally several points:
+
+- Technical issues related to the initial transition and subsequent maintenance of *Black* formatting are surmountable.
+  There is plenty of prior art and good documentation.
+- Sociological issues related to developer preferences are surmountable.
+  Evidence from other projects is that developers either embrace or at least accept autoformatting.
+- Developer documentation can provide clear instructions to contributors on how to implement *Black* formatting in their development process.
+  Issues with this process are no more difficult (and arguably easier) to resolve than enforcing PEP-8 standards.
+
+Therefore, this SEP proposes to adopt *Black* for the SunPy Project.
+
+#### Concerns and mitigations
+
+##### Foolish consistency
+
+A concern that has been highlighted by some members of the community (including, initially, the author of this SEP, and highlighted in detail in [this blog post](https://luminousmen.com/post/my-unpopular-opinion-about-black-code-formatter) is that adopting *Black* violates the PEP-8 principle that "A Foolish Consistency is the Hobgoblin of Little Minds".
+I.e., that by using a code formatter, the developer thinks less about readability because they think they can let the formatter do this for them.
+While this concern has some validity, this SEP explicitly recognizes that **developers and code reviewers are ultimately responsible for maintaining a readable code base**.
+That is, there are many choices concerning logic and readability of code that *Black* does *not* affect, and those choices are more important for developers to focus on than mechanistic decisions that *Black* makes about whitespace and newlines.
+
+##### Undesirable formatting
+
+On occasion *Black*'s formatting produces undesirable results.
+The code may look very different from expectation due to mathematical or other domain-specific conventions, or some specific formatting that conveys meaning may be lost.
+A simple example of this is a constant identity matrix which would typically be written as
+
+```python
+    matrix = [[1, 0, 0],
+              [0, 1, 0],
+              [0, 0, 1]]
+```
+
+*Black* reformats this as shown below in a way that is very different from the typical representation for a matrix and makes it harder to read the structure
+
+```python
+    matrix = [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
+```
```

Comment:
Entertainingly this example will be formatted differently if you include a trailing comma on the last row.

Reply:
Removed.